File: CrowdfundPro/crowdfundpro_frontend/src/types/project.ts
export interface Project {
  id: number;
  titre: string;
  description: string;
  image_url?: string;
  objectif: number;
  montant_actuel: number;
  montant_requis: number;
  pourcentage_finance: number;
  nombre_investisseurs: number;
  jours_restants: number;
  categorie: string;
  date_creation: string;
  date_limite: string;
  statut: 'EN_ATTENTE_VALIDATION' | 'EN_COURS' | 'FINANCE' | 'ECHOUE';
  statut_display?: string;
  porteur: {
    id: number;
    username: string;
    nom: string;
    email: string;
    date_inscription: string;
  };
  // Nouveaux champs pour la localisation
  latitude?: number;
  longitude?: number;
  adresse?: string;
  a_localisation?: boolean;
  // Nouveaux champs pour les documents
  business_plan?: string;
  plan_juridique?: string;
} 

File: CrowdfundPro/MyApplication3/app/src/main/java/com/example/myapplication/data/model/Project.kt
package com.example.myapplication.data.model

import com.google.gson.annotations.SerializedName

data class Project(
    val id: Int,
    val titre: String,
    val description: String,
    val objectif: Double,
    @SerializedName("montant_actuel")
    val montantActuel: Double,
    @SerializedName("date_creation")
    val dateCreation: String,
    @SerializedName("date_limite")
    val dateLimite: String,
    val statut: ProjectStatus,
    @SerializedName("statut_display")
    val statutDisplay: String,
    val image: String? = null,
    val porteur: User,
    @SerializedName("nombre_investisseurs")
    val nombreInvestisseurs: Int = 0,
    @SerializedName("pourcentage_finance")
    val pourcentageFinance: Double = 0.0,
    @SerializedName("montant_cible")
    val montantCible: Double = 0.0,
    @SerializedName("date_debut")
    val dateDebut: String? = null,
    @SerializedName("date_fin")
    val dateFin: String? = null,
    val investissements: List<Investment>? = null,
    val latitude: Double? = null,
    val longitude: Double? = null,
    val adresse: String? = null,
    @SerializedName("a_localisation")
    val aLocalisation: Boolean = false,
    @SerializedName("business_plan")
    val businessPlan: String? = null,
    @SerializedName("plan_juridique")
    val planJuridique: String? = null
)

enum class ProjectStatus {
    @SerializedName("EN_ATTENTE_VALIDATION")
    EN_ATTENTE_VALIDATION,
    @SerializedName("EN_COURS")
    EN_COURS,
    @SerializedName("FINANCE")
    FINANCE,
    @SerializedName("ECHOUE")
    ECHOUE
}

data class ProjectCreateRequest(
    val titre: String,
    val description: String,
    val objectif: Double,
    @SerializedName("date_limite")
    val dateLimite: String,
    val latitude: Double? = null,
    val longitude: Double? = null,
    val adresse: String? = null
)

data class ProjectStats(
    val total: Int,
    @SerializedName("en_cours")
    val enCours: Int,
    val finance: Int,
    val echoue: Int,
    @SerializedName("montant_total")
    val montantTotal: Double,
    @SerializedName("montant_moyen")
    val montantMoyen: Double
)

data class PaginatedResponse<T>(
    val count: Int,
    val next: String?,
    val previous: String?,
    val results: List<T>
) 